sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt

```diff
@@ -366,11 +380,17 @@ internal class KlaviyoLocationManager : LocationManager {
                     return@mapNotNull null
                 }
 
-                Registry.log.info("Triggered geofence $geofenceTransition $kGeofence")
-
                 // Create the event and record the transition time
                 cooldownTracker.recordTransition(kGeofence.id, geofenceTransition)
-                createGeofenceEvent(geofenceTransition, kGeofence)
+
+                // Look up the stored geofence to get the duration
+                val modifiedGeofence = storedGeofences.find { it.id == kGeofence.id }?.let { storedGeofence ->
+                    kGeofence.copy(duration = storedGeofence.duration)
```

Comment:
This is just copying over the duration value that was configured on the stored geofence... I suppose that's an OK fallback.
But if Android tells use the actual duration value that was observed to trigger this event, that'd be better.